### ISSUE TITLE
Downloading modules using by getall script fails

### ISSUE THREAD

**Issue by stefanhusmann** [FreeFem/FreeFem-sources#124]
I think there is a problem on the server from which getall pulls the third party files. 

' ' ' 
Connecting to pkgs.freefem.org (pkgs.freefem.org)|134.157.2.3|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 4804966 (4.6M) [application/x-gzip]
Saving to: 'pkg/scotch_6.0.4.tar.gz'

pkg/scotch_6.0.4.tar.g   0%[                             ]       0  --.-KB/s               
pkg/scotch_6.0.4.tar.g  20%[====>                        ] 953.14K  4.65MB/s               
pkg/scotch_6.0.4.tar.g  50%[=============>               ]   2.33M  5.82MB/s               
pkg/scotch_6.0.4.tar.g  81%[======================>      ]   3.72M  6.21MB/s               
pkg/scotch_6.0.4.tar.g 100%[============================>]   4.58M  6.34MB/s    in 0.7s    

2020-01-12 19:53:19 (6.34 MB/s) - 'pkg/scotch_6.0.4.tar.gz' saved [4804966/4804966]

 incorrect file type => removing  pkg/scotch_6.0.4.tar.gz

ERROR: ../bin/build/download http://104.46.50.187/pkg/scotch_6.0.4.tar.gz pkg/scotch_6.0.4.tar.gz BAD_CERT FAILED
ERROR 2: ../bin/build/download http://pkgs.freefem.org/scotch_6.0.4.tar.gz pkg/scotch_6.0.4.tar.gz BAD_CERT FAILED
ERROR2: INVALID MD5 for scotch_6.0.4.tar.gz  d41d8cd98f00b204e9800998ecf8427e
```

**Comment by prj-** [comment]
I'd suggest that you compile FreeFEM with PETSc support (`cd 3rdparty/ff-petsc && make petsc-slepc`), it will fetch the proper packages which will be used afterwards, even if you don't use PETSc in your scripts.

**Comment by stefanhusmann** [comment]
That is not an option. I also do not see how this should fix md5sum errors.

**Comment by prj-** [comment]
It will fix the error, as it won't use this outdated SCOTCH version. It will use an updated version instead.

**Comment by stefanhusmann** [comment]
It is not an option because the build process tries to place some files in systemwide directories, and setting DESTDIR does not help in this respect - and it even needs sudo to be installed, which I cannot rely on. I want to do this in an AUR-package for Arch Linux.

**Comment by prj-** [comment]
That's not true. Just use the --prefix=${PWD} option. You won't have any systemwide directories then.

**Comment by stefanhusmann** [comment]
```sh
make --prefix=$PWD petsc-slepc
make: unrecognized option '--prefix=$PWD'
Usage: make [options] [target] ...
```
and
```sh
make PREFIX=$PWD petsc-slepc
mkdir -p /usr/ff-petsc//real
mkdir: cannot create directory '/usr/ff-petsc': Permission denied
make: [Makefile:197: dir-install] Error 1 (ignored)
chown haawda /usr/ff-petsc//real
chown: cannot access '/usr/ff-petsc//real': No such file or directory
make: [Makefile:198: dir-install] Error 1 (ignored)
mkdir -p /usr/ff-petsc//complex
mkdir: cannot create directory '/usr/ff-petsc': Permission denied
make: [Makefile:199: dir-install] Error 1 (ignored)
chown haawda /usr/ff-petsc//complex
chown: cannot access '/usr/ff-petsc//complex': No such file or directory
make: [Makefile:200: dir-install] Error 1 (ignored)
no write access in /usr/ff-petsc//real and /usr/ff-petsc//complex
 do: make petsc-slepc SUDO=sudo 
 or add write access to 2 directory 
test -e "/usr/ff-petsc//real" -a -e "/usr/ff-petsc//complex"
make: *** [Makefile:202: dir-install] Error 1
```

**Comment by stefanhusmann** [comment]
Okay DESTIR=. works, but then I get md5sum-errors for petsc-lite
```sh

--2020-01-22 20:12:30--  http://ftp.mcs.anl.gov/pub/petsc/release-snapshots/petsc-lite-3.12.3.tar.gz
Resolving ftp.mcs.anl.gov (ftp.mcs.anl.gov)... 140.221.6.23
Connecting to ftp.mcs.anl.gov (ftp.mcs.anl.gov)|140.221.6.23|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 14228267 (14M) [text/plain]
Saving to: 'pkg/petsc-lite-3.12.3.tar.gz'


pkg/petsc-lite-3.12.3.tar   0%[                                     ]       0  --.-KB/s               
pkg/petsc-lite-3.12.3.tar   0%[                                     ]  43.35K   193KB/s               
pkg/petsc-lite-3.12.3.tar   1%[                                     ] 212.10K   471KB/s               
pkg/petsc-lite-3.12.3.tar   6%[=>                                   ] 881.48K  1.28MB/s               
pkg/petsc-lite-3.12.3.tar  16%[=====>                               ]   2.27M  2.60MB/s               
pkg/petsc-lite-3.12.3.tar  27%[========>                            ]   3.67M  3.41MB/s               
pkg/petsc-lite-3.12.3.tar  37%[============>                        ]   5.06M  3.97MB/s               
pkg/petsc-lite-3.12.3.tar  47%[================>                    ]   6.46M  4.38MB/s               
pkg/petsc-lite-3.12.3.tar  57%[====================>                ]   7.85M  4.69MB/s               
pkg/petsc-lite-3.12.3.tar  65%[=======================>             ]   8.87M  4.50MB/s               
pkg/petsc-lite-3.12.3.tar  83%[=============================>       ]  11.27M  5.20MB/s               
pkg/petsc-lite-3.12.3.tar  91%[================================>    ]  12.45M  5.23MB/s               
pkg/petsc-lite-3.12.3.tar  99%[===================================> ]  13.53M  5.25MB/s               
pkg/petsc-lite-3.12.3.tar 100%[====================================>]  13.57M  5.25MB/s    in 2.6s    

2020-01-22 20:12:33 (5.25 MB/s) - 'pkg/petsc-lite-3.12.3.tar.gz' saved [14228267/14228267]

 incorrect file type => removing  pkg/petsc-lite-3.12.3.tar.gz
Download failed from http://ftp.mcs.anl.gov/pub/petsc/release-snapshots/petsc-lite-3.12.3.tar.gz of petsc-lite-3.12.3.tar.gz
Try other site: http://pkgs.freefem.org/petsc-lite-3.12.3.tar.gz
--2020-01-22 20:12:33--  http://pkgs.freefem.org/petsc-lite-3.12.3.tar.gz
Resolving pkgs.freefem.org (pkgs.freefem.org)... 134.157.2.3
Connecting to pkgs.freefem.org (pkgs.freefem.org)|134.157.2.3|:80... connected.
HTTP request sent, awaiting response... 404 Not Found
2020-01-22 20:12:34 ERROR 404: Not Found.

Error download pkg/petsc-lite-3.12.3.tar.gz
Download 2 times failed from http://pkgs.freefem.org/petsc-lite-3.12.3.tar.gz of petsc-lite-3.12.3.tar.gz
Try (2 times)  other site: http://104.46.50.187/pkg/petsc-lite-3.12.3.tar.gz
--2020-01-22 20:12:34--  http://104.46.50.187/pkg/petsc-lite-3.12.3.tar.gz
Connecting to 104.46.50.187:80... connected.
HTTP request sent, awaiting response... 404 Not Found
2020-01-22 20:12:34 ERROR 404: Not Found.

Error download pkg/petsc-lite-3.12.3.tar.gz
Download failed (MD5 check) from http://ftp.mcs.anl.gov/pub/petsc/release-snapshots/petsc-lite-3.12.3.tar.gz of petsc-lite-3.12.3.tar.gz
Try other site: http://pkgs.freefem.org/petsc-lite-3.12.3.tar.gz
--2020-01-22 20:12:34--  http://pkgs.freefem.org/petsc-lite-3.12.3.tar.gz
Resolving pkgs.freefem.org (pkgs.freefem.org)... 134.157.2.3
Connecting to pkgs.freefem.org (pkgs.freefem.org)|134.157.2.3|:80... connected.
HTTP request sent, awaiting response... 404 Not Found
2020-01-22 20:12:34 ERROR 404: Not Found.

Error download pkg/petsc-lite-3.12.3.tar.gz

ERROR: ../bin/build/download http://104.46.50.187/pkg/petsc-lite-3.12.3.tar.gz pkg/petsc-lite-3.12.3.tar.gz  FAILED
ERROR 2: ../bin/build/download http://pkgs.freefem.org/petsc-lite-3.12.3.tar.gz pkg/petsc-lite-3.12.3.tar.gz  FAILED
ERROR2: INVALID MD5 for petsc-lite-3.12.3.tar.gz  d41d8cd98f00b204e9800998ecf8427e
make: *** [Makefile:328: ../pkg/petsc-lite-3.12.3.tar.gz] Error 1
==> ERROR: A failure occurred in build().
    Aborting...

Process makepkg exited abnormally with code 4
```

**Comment by prj-** [comment]
I've downloaded `http://ftp.mcs.anl.gov/pub/petsc/release-snapshots/petsc-lite-3.12.3.tar.gz` on 4 different machines and I've got the correct MD5 hash on all of them, `ca5c8f535ff93e22a0bd2e03ded4befe`. I don't really know what is up with your MD5 hash calculator, but I guess this is not a FreeFEM problem and rather something being inconsistent on your machine. By the way, the `--prefix` option must be supplied at FreeFEM `./configure` time, but anyway, this won't fix the incorrect hashes.
